mongo/options/autoencryptionoptions.go

```diff
@@ -143,6 +147,14 @@ func (a *AutoEncryptionOptions) SetTLSConfig(tlsOpts map[string]*tls.Config) *Au
 	return a
 }
 
+// SetHTTPClient specifies the http.Client to be used for any KMS provider making HTTP requests.
+//
+// This should only be used to set custom HTTP client configurations. By default, the connection will use a http.DefaultClient.
+func (a *AutoEncryptionOptions) SetHTTPClient(client *http.Client) *AutoEncryptionOptions {
```

Comment:
What is the use case for supplying a custom HTTP client for calling KMS providers?

Reply:
you may have custom logic in your transport

Reply:
in general, most popular go packages that use HTTP either let you specify your own http.Client or Transport.